transform/app/services/s3/get_s3_paths.py

```diff
@@ -43,34 +43,51 @@ def get_s3_paths(
-                        file_key, key, s3_directory
-                    ) or is_exact_directory_match(
-                        file_key, entity_mapping.get(key, key), s3_directory
-                    ):
-                        if file_key.endswith(".zip"):
-                            zip_content = BytesIO(
-                                s3_client.get_object(Bucket=bucket_name, Key=file_key)[
-                                    "Body"
-                                ].read()
-                            )
-                            files_dict[key] = [
-                                f"{settings.S3_ENDPOINT}files/{bucket_name}/{s3_directory}/{zip_file}"
-                                for zip_file in list_files_in_zip(zip_content)
-                            ]
-                        elif (
-                            file_key
-                            != f"{s3_directory}/{entity_mapping.get(key, key)}/"
+        continuation_token = None
+        while True:
+            response = (
+                s3_client.list_objects_v2(
+                    Bucket=bucket_name,
+                    Prefix=s3_directory,
+                    ContinuationToken=continuation_token,
+                )
+                if continuation_token
+                else s3_client.list_objects_v2(Bucket=bucket_name, Prefix=s3_directory)
+            )
+
+            if "Contents" in response:
+                for obj in response["Contents"]:
+                    file_key = obj["Key"]
+                    for key in files_dict.keys():
+                        if is_exact_directory_match(
+                            file_key, key, s3_directory
+                        ) or is_exact_directory_match(
+                            file_key, entity_mapping.get(key, key), s3_directory
                         ):
-                            files_dict[key].append(
-                                f"{settings.S3_ENDPOINT}files/{bucket_name}/{file_key}"
-                            )
-                        break
+                            if file_key.endswith(".zip"):
+                                zip_content = BytesIO(
+                                    s3_client.get_object(
+                                        Bucket=bucket_name, Key=file_key
+                                    )["Body"].read()
+                                )
+                                files_dict[key] = [
+                                    f"{settings.S3_ENDPOINT}files/{bucket_name}/{s3_directory}/{zip_file}"
+                                    for zip_file in list_files_in_zip(zip_content)
+                                ]
+                            elif (
+                                file_key
+                                != f"{s3_directory}/{entity_mapping.get(key, key)}/"
+                            ):
+                                files_dict[key].append(
+                                    f"{settings.S3_ENDPOINT}files/{bucket_name}/{file_key}"
```

Comment:
also make it more robust, ensure settings.S3_ENDPOINT ends with a '/'. E.g. or use rstrip 
`s3_base_url = settings.S3_ENDPOINT if settings.S3_ENDPOINT.endswith('/') else f"{settings.S3_ENDPOINT}/"`

Reply:
Changed